### PR TITLE
Implement cache policy "NoCache"

### LINKED ISSUE #356: Provide API for fetching data without caching the result
In situations like search, the value doesn't always need to be cached. We should give users a way to fetch, paginate, etc. without writing the values to cache

### PR DESCRIPTION
**Pull request by MattiasMartens**
Fixes #356

Implement the policy "NoCache", mirroring Apollo, which serves a request without reading from or writing to the cache.

Tests are failing for me locally with messages like this:

```
TypeError: Cannot read properties of undefined (reading 'on')
 ❯ vitest.setup.ts:12:1
     10| process.env.HOUDINI_TEST = 'true'
     11| 
     12| beforeEach(clearMock)
       | ^
     13| 
     14| const config = testConfig()
```

This happens even without my changes, but it means I'm not able to write tests for the change.

### To help everyone out, please make sure your PR does the following:

- [X] Update the first line to point to the ticket that this PR fixes
- [X] Add a message that clearly describes the fix
~~- [ ] If applicable, add a test that would fail without this fix
~~- [ ] Make sure the unit and integration tests pass locally with `pnpm run tests` and `cd integration && pnpm run tests`
- [X ] Includes a changeset if your fix affects the user with `pnpm changeset`

### CODE REVIEW

File: packages/houdini/src/runtime/client/documentStore.ts [hunk 1]
```diff
@@ -200,6 +200,8 @@ export class DocumentStore<
 		getSession: () => App.Session | null | undefined
 	): (input: RequestInfo | URL, init?: RequestInit | undefined) => Promise<Response> {
 		return async (input, init) => {
+			// FINDME whatever calls this is what will decide whether/how to cache it.
```
Comment (AlecAivazis) [inline packages/houdini/src/runtime/client/documentStore.ts:203]: I'm a bit confused what the FINDME comments are referring to here. the caching logic should be totally confined to the `cache` plugin

**Review by AlecAivazis — CHANGES_REQUESTED** [review]
Changes look mostly good! Should be able to merge this in soon.

File: packages/houdini/src/runtime/client/documentStore.ts (hunk 1 above)
Comment (MattiasMartens) [reply packages/houdini/src/runtime/client/documentStore.ts:203]: Oh no! Trivial mistake on my part. Those were notes I left as I navigated the code. I will remove them.

**Comment by MattiasMartens** [comment]
@AlecAivazis Great! Test snapshots are now updated.

File: packages/houdini/src/codegen/transforms/schema.ts [hunk 2]
```diff
@@ -18,6 +18,7 @@ enum CachePolicy {
 	${CachePolicy.CacheOnly}
 	${CachePolicy.CacheOrNetwork}
 	${CachePolicy.NetworkOnly}
+  ${CachePolicy.NoCache}
```
Comment (SeppahBaws) [inline packages/houdini/src/codegen/transforms/schema.ts:21]: pretty sure this line will fail the formatting check, due to mismatching indentations
`pnpm format` should do the trick 🙂

**Review by SeppahBaws — COMMENTED** [review]
Should we maybe also add a unit test to ensure that we're 100% certain it doesn't write to the cache?

**Comment by MattiasMartens** [comment]
@SeppahBaws As I said above, I am not able to write or run tests for this branch! I get this error for most tests:

```
TypeError: Cannot read properties of undefined (reading 'on')
 ❯ vitest.setup.ts:12:1
     10| process.env.HOUDINI_TEST = 'true'
     11| 
     12| beforeEach(clearMock)
       | ^
     13| 
     14| const config = testConfig()
```

**Comment by SeppahBaws** [comment]
@MattiasMartens yes of course, was more of a question in general :)
I can take a look at writing a test when I have some spare time, or maybe @AlecAivazis can if he can find the time before I do.

**Comment by AlecAivazis** [comment]
I had a few moments tonight so I pushed up a test 👍

**Review by AlecAivazis — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**